client/components/header-cake/back.jsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import ObserveWindowSizeMixin from 'lib/mixins/observe-window-resize';
+import Gridicon from 'components/gridicon';
+
+/**
+ * Module variables
+ */
+const HIDE_BACK_CRITERIA = {
+	windowWidth: 480,
+	characterLength: 8
+};
+
+export default React.createClass( {
+
+	displayName: 'HeaderCakeBack',
+
+	mixins: [ ObserveWindowSizeMixin ],
+
+	propTypes: {
+		onClick: PropTypes.func,
+		href: PropTypes.string,
+		text: PropTypes.string
+	},
+
+	getDefaultProps() {
+		return {
+			text: 'Back'
+		}
+	},
+
+	render() {
+		const { text, href, onClick } = this.props;
+		const hideText = window.innerWidth <= HIDE_BACK_CRITERIA.windowWidth && text.length >= HIDE_BACK_CRITERIA.characterLength;
+
+		return (
+			<a className="header-cake__back" href={ href } onClick={ onClick }>
+				<Gridicon icon="chevron-left" size={ 18 } />
+				{ hideText ?
+					null
+				: <span className="header-cake__back-text">{ text }</span> }
```

Comment:
This looks a bit off here with the `null` and the alignment. We have been using `&&` when we only have one display condition. It might help here.

```js
{ ! hideText && <span className="header-cake__back-text">{ text }</span> }
```

Of course, then it might make more sense to name the condition in the positive…

```js
const showText = window.innerWidth > HIDE_BACK_CRITERIA.windowWidth && text.length < HIDE_BACK_CRITERIA.characterLength;
```